opentelemetry-api/src/opentelemetry/util/_importlib_metadata.py

```diff
@@ -12,18 +12,30 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# FIXME: Use importlib.metadata when support for 3.11 is dropped if the rest of
-# the supported versions at that time have the same API.
-from importlib_metadata import (  # type: ignore
-    EntryPoint,
-    EntryPoints,
-    entry_points,
-    version,
-)
-
 # The importlib-metadata library has introduced breaking changes before to its
 # API, this module is kept just to act as a layer between the
 # importlib-metadata library and our project if in any case it is necessary to
 # do so.
 
+from sys import version_info
+
+if version_info.minor < 10:
```

Comment:
It would be helpful to add a comment somewhere because we've returned this topic a few times in the sig meetings and have to re-contextualize, [example issue](https://github.com/open-telemetry/opentelemetry-python/pull/4144#issuecomment-2305254753). Maybe something like:

```suggestion
# Python 3.10 introduces feature parity with `importlib_metadata` via `importlib.metadata`
# https://docs.python.org/3/whatsnew/3.10.html#importlib-metadata
if version_info.minor < 10:
```

Reply:
Right, added a comment that explains this better.